state/txindex/kv/kv.go

```diff
@@ -40,6 +47,82 @@ type TxIndex struct {
 	log log.Logger
 }
 
+func (txi *TxIndex) Prune(retainHeight int64) (int64, int64, error) {
+	// Returns the last retained height
+	lastRetainHeight, err := txi.getLastTxIndexerRetainHeight()
+	if err != nil {
+		return 0, 0, fmt.Errorf("failed to look up last block indexer retain height: %w", err)
+	}
+	if lastRetainHeight == 0 {
+		lastRetainHeight = 1
+	}
+
+	ctx := context.Background()
+	results, err := txi.Search(ctx, query.MustCompile(
+		fmt.Sprintf("tx.height < %d AND tx.height >= %d", retainHeight, lastRetainHeight)))
+	if err != nil {
+		panic(err)
+	}
+	if len(results) == 0 {
+		return 0, lastRetainHeight, nil
+	}
+
+	sort.Sort(TxResultByHeight(results))
```

Comment:
I am not sure how expensive this is. I understand you need it because you need to determine the lastRetainHeight. But in theory, if the result was returned by the Search then it should also be deletable .

Reply:
Does the Search provide results sorted?

Reply:
My logic was this is not expansive, since sorts work in _O(n*log(n))_ where n is the number of results. And we pay at least a cost linear in _n_ , so not that much of an overhead

Reply:
No they are not sorted so this is ok the way it is. We really do hope the indexer will go away eventually :)